bndtools.core/resources/unprocessed/serviceTemplates/java-service-api/bnd.bnd

```diff
@@ -0,0 +1,11 @@
+$if(javaLevel)$
+javac.source=$javaLevel$
+javac.target=$javaLevel$
+javac.compliance=$javaLevel$
+$endif$
+
+-buildpath: \
+	osgi.annotation,\
+	org.osgi.annotation.bundle
+	
+Bundle-Version: 1.0.0.\${tstamp}
```

Comment:
I do not think it is a good idea to set the Bundle-Version in the bnd file. I find that releasing the whole workspace with one version is better.

Reply:
I don't understand what you mean by '...general setup stuff.  I think we should keep that by default in the workspace build.bnd'.   Doesn't this make everything bound to that one workspace?   In other words...for Bundle-Version and other stuff...where can it go so that a specific workspace isn't required? (i.e. or in all workspaces)?

Reply:
Yes. The default mode for bndtools is that your workspace is a cohesive set of bundles. This deviates from the Eclipse model where your workspace is an unrelated set of things. That is why a workspace for us is in one directory.

Since the tooling tends to do a release version on the workspace, you're seriously screwed when you keep the versions in the bnd files since you cannot release the same version twice. So all the workspace I'm responsible for are keeping the bundle version in the build.bnd. Saves a lot of pain! :-) The cost that identical bundles appear with multiple versions is something will inevitably cause the world to heat up faster but somehow can't keep me awake at night :-)

A bnd workspace is a bit like the Reactor (?) model of maven. It is a bunch of models that are cohesive.

Reply:
So just to be clear, are you saying that I should just remove the Bundle-Version header from the templates and expect that it will be provided by the cnf/build.bnd for all projects in the workspace?

Reply:
removed